src/uproot/_dask.py

```diff
@@ -731,142 +743,302 @@ def _get_dask_array_delay_open(
+    ) -> Mapping[str, AwkArray]:
+        ...
+
+
+class ImplementsFormMapping(Protocol):
+    def __call__(self, form: Form) -> tuple[Form, ImplementsFormMappingInfo]:
+        ...
+
+
+class TrivialFormMappingInfo(ImplementsFormMappingInfo):
+    def __init__(self, form):
+        awkward = uproot.extras.awkward()
+        assert isinstance(form, awkward.forms.RecordForm)
+
+        self._form = form
+        self._form_key_to_key = self.build_form_key_to_key(form)
+
+    @property
+    def behavior(self) -> None:
+        return None
+
+    @staticmethod
+    def build_form_key_to_key(form: Form) -> dict[str, str | None]:
+        form_key_to_path: dict[str, str | None] = {}
+
+        def impl(form, column_path):
+            # Store columnar path
+            form_key_to_path[form.form_key] = column_path[-1] if column_path else None
+
+            if form.is_union:
+                for _i, entry in enumerate(form.contents):
+                    impl(entry, column_path)
+            elif form.is_indexed:
+                impl(form.content, column_path)
+            elif form.is_list:
+                impl(form.content, column_path)
+            elif form.is_option:
+                impl(form.content, column_path)
+            elif form.is_record:
+                for field in form.fields:
+                    impl(form.content(field), (*column_path, field))
+            elif form.is_unknown or form.is_numpy:
+                pass
             else:
-                actual_form = self.rendered_form
+                raise AssertionError(form)
 
-            mapping, buffer_key = self.form_mapping.create_column_mapping_and_key(
-                self.ttrees[i], start, stop, self.interp_options
-            )
+        impl(form, ())
 
-            layout = awkward.from_buffers(
-                actual_form,
-                stop - start,
-                mapping,
-                buffer_key=buffer_key,
-                highlevel=False,
-            )
+        return form_key_to_path
 
-            return awkward.Array(
-                dask_awkward.lib.unproject_layout.unproject_layout(
-                    self.rendered_form,
-                    layout,
-                ),
-                behavior=self.form_mapping.behavior,
-            )
+    buffer_key: Final[str] = "{form_key}-{attribute}"
+
+    def parse_buffer_key(self, buffer_key: str) -> tuple[str, str]:
+        form_key, attribute = buffer_key.rsplit("-", maxsplit=1)
+        return form_key, attribute
 
-        array = self.ttrees[i].arrays(
-            self.common_keys,
+    def keys_for_buffer_keys(self, buffer_keys: set[str]) -> set[str]:
+        keys: set[str] = set()
+        for buffer_key in buffer_keys:
+            # Identify form key
+            form_key, attribute = buffer_key.rsplit("-", maxsplit=1)
+            # Identify key from form_key
+            keys.add(self._form_key_to_key[form_key])
+        return keys
```

Comment:
As per the new dask-awkward buffer projection, we implement a function that maps buffer keys (of our own choosing!) to the TTree keys.